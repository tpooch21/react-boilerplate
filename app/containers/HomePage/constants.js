/**
 * HOME PAGE constants
 */

export const LOAD_STRINGS = 'stringDisplayer/HomePage/LOAD_STRINGS';
export const LOAD_STRINGS_SUCCESS =
  'stringDisplayer/HomePage/LOAD_STRINGS_SUCCESS';
