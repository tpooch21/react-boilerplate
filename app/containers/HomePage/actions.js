import { LOAD_STRINGS, LOAD_STRINGS_SUCCESS } from './constants';

export const loadStrings = () => ({
  type: LOAD_STRINGS
});

export const loadStringsSuccess = () => ({
  type: LOAD_STRINGS_SUCCESS,
  strings,
});
