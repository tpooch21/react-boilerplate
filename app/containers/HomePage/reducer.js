import produce from 'immer';

import { LOAD_STRINGS, LOAD_STRINGS_SUCCESS } from './constants';

export const initialState = {
  strings: [],
  loading: false,
  error: false,
};

const homePageReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case LOAD_STRINGS:
        draft.loading = true;
        draft.error = false;
        break;
      case LOAD_STRINGS_SUCCESS:
        draft.loading = false;
        draft.error = false;
        draft.strings = action.strings;
        break;
    }
  });

export default homePageReducer;
