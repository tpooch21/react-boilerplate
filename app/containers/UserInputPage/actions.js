import { ENTER_STRING } from './constants';

export const enterString = input => ({
  type: ENTER_STRING,
  currentString: input,
});
