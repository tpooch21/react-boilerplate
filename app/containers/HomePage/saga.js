import { call, put, select, takeLatest } from 'redux-saga/effects';
import { axiosReq } from 'utils/axios-config';
import { LOAD_STRINGS } from './constants';

import { loadStringsSuccess } from './actions';

export function* getStrings() {
  const url = 'http://localhost:3000';
  console.log('We in here?');

  try {
    const strings = yield call(axiosReq, url);
    console.log('Logging strings returned from API => ', strings);
  } catch (err) {
    console.log('Logging error on string fetch => ', err);
  }
}

export default function* getStringData() {
  yield takeLatest(LOAD_STRINGS, getStrings);
}
