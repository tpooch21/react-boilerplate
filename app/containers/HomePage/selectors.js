import { createSelector } from 'reselect';
import { initialState } from './reducer';

// Slice state properties that concern HomePage
export const selectHomePage = state => state.homepage || initialState;

export const makeSelectStrings = () =>
  createSelector(
    selectHomePage,
    homeState => homeState.strings,
  );

export const makeSelectLoading = () =>
  createSelector(
    selectHomePage,
    homeState => homeState.loading,
  );

export const makeHomeSelectError = () =>
  createSelector(
    selectHomePage,
    homeState => homeState.error,
  );
