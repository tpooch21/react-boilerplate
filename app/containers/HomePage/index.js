/*
 * HomePage
 *
 * This is the first thing users see of our App, at the '/' route
 *
 */

import React, { useEffect } from 'react';
import { FormattedMessage } from 'react-intl';
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';
import { connect } from 'react-redux';
import { compose } from 'redux';
import PropTypes from 'prop-types';
import { createStructuredSelector } from 'reselect';
import reducer from './reducer';
import * as actions from './actions';
import messages from './messages';
import saga from './saga';
import HomeWrapper from '../../components/Wrapper/index';
import { makeSelectStrings, makeSelectLoading } from './selectors';

import StringDisplay from '../../components/StringDisplay/index';

const HomePage = props => {
  useEffect(() => {
    props.loadStrings();
  }, []);

  return (
    <HomeWrapper>
      <h2>
        <FormattedMessage {...messages.header} />
      </h2>
      <StringDisplay strings={props.strings} loading={props.loading} />
    </HomeWrapper>
  );
};

const mapStateToProps = createStructuredSelector({
  strings: makeSelectStrings(),
  loading: makeSelectLoading(),
});

export const mapDispatchToProps = dispatch => ({
  loadStrings: () => dispatch(actions.loadStrings()),
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'homepage', reducer });

const withSaga = injectSaga({ key: 'homepage', saga });

HomePage.propTypes = {
  loadStrings: PropTypes.func,
  strings: PropTypes.array,
  loading: PropTypes.bool,
};

export default compose(
  withSaga,
  withReducer,
  withConnect,
)(HomePage);
