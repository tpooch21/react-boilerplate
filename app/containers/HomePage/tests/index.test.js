import React from 'react';
import { render } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import HomePage, { mapDispatchToProps } from '../index';
import configureStore from '../../../configureStore';
import { loadStrings } from '../actions';

describe('<HomePage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <HomePage />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('mapDispatchToProps', () => {
    describe('loadStrings', () => {
      it('should be injected', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        expect(result.loadStrings).toBeDefined();
      });

      it('should dispatch loadStrings action when called', () => {
        const dispatch = jest.fn();
        const result = mapDispatchToProps(dispatch);
        result.loadStrings();
        expect(dispatch).toHaveBeenCalledWith(loadStrings());
      });
    });
  });
});
