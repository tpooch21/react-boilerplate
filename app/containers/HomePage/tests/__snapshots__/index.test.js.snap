// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should render and match the snapshot 1`] = `
<div
  class="Wrapper-tno06a-0 bqIzVi"
>
  <h2>
    <span>
      View Your Words and Phrases Below
    </span>
  </h2>
  <div
    class="spinner"
  >
    <div
      class="rect1"
    />
    <div
      class="rect2"
    />
    <div
      class="rect3"
    />
    <div
      class="rect4"
    />
    <div
      class="rect5"
    />
  </div>
</div>
`;
